### PR TITLE
Fix nightly tests and bump version 0.18.dev0

### PR DESCRIPTION
**Pull request by mkundu1**
Addressing the failing tests in https://github.com/ansys/pyfluent/actions/runs/6020381003/job/16331606474#step:22:3080

### CODE REVIEW

File: tests/test_solvermode/test_initialization.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 @pytest.mark.nightly
 @pytest.mark.quick
 @pytest.mark.setup
-@pytest.mark.fluent_version("latest")
+@pytest.mark.fluent_version("==23.2")
```
Comment (mkundu1) [inline tests/test_solvermode/test_initialization.py:8]: This test compares settings state dict and is too sensitive to API change. I don't think we need to test it with the dev version.

**Review by raph-luc — APPROVED** [review]
For information, I noticed that the nightly tests for this branch are currently queued to run here: https://github.com/ansys/pyfluent/actions/runs/6027401498/job/16352524434

**Review by hpohekar — APPROVED** [review]

**Comment by mkundu1** [comment]
> For information, I noticed that the nightly tests for this branch are currently queued to run here: https://github.com/ansys/pyfluent/actions/runs/6027401498/job/16352524434

Yeah, I suspect some 23.2 nightly tests will still be failing, will address them after getting the 23.2 results (which are run only if 24.1 is successful)

**Comment by mkundu1** [comment]
The nightly run keeps failing, although the individual tests are running OK locally. Merging this now, will check the next official nightly run.

**Pull request merged**